src/data/java_profile.rs

```diff
@@ -0,0 +1,332 @@
+extern crate ctor;
+
+use crate::data::{CollectData, CollectorParams, Data, DataType, ProcessedData};
+use crate::visualizer::{DataVisualizer, GetData};
+use crate::{PERFORMANCE_DATA, VISUALIZATION_DATA};
+use anyhow::Result;
+use ctor::ctor;
+use log::{debug, error, trace};
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use std::io::{ErrorKind, Write};
+use std::path::PathBuf;
+use std::process::{Child, Command};
+use std::sync::Mutex;
+use std::{fs, fs::File};
+
+pub static JAVA_PROFILE_FILE_NAME: &str = "java_profile";
+
+lazy_static! {
+    pub static ref ASPROF_CHILDREN: Mutex<Vec<Child>> = Mutex::new(Vec::new());
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct JavaProfileRaw {
+    process_map: HashMap<String, Vec<String>>,
+}
+
+impl JavaProfileRaw {
+    pub fn new() -> Self {
+        JavaProfileRaw {
+            process_map: HashMap::new(),
+        }
+    }
+
+    fn launch_asprof(&self, jids: Vec<String>, params: CollectorParams) -> Result<()> {
+        let data_dir = PathBuf::from(params.data_dir.clone());
+        for jid in &jids {
+            let mut html_loc = data_dir.clone();
+            html_loc.push(format!("{}-java-flamegraph-{}.html", params.run_name, jid));
+
+            match Command::new("asprof")
+                .args([
+                    "-d",
+                    &(params.collection_time - params.elapsed_time).to_string(),
+                    "-f",
+                    html_loc.to_str().unwrap(),
+                    jid.as_str(),
+                ])
+                .spawn()
+            {
+                Err(e) => {
```

Comment:
Do we want to fail here? The user java profiling but we are unable to do it. This might be also applicable to perf. Maybe something common could be done. It shouldn't be a blocker for now.

Reply:
Also, we print this message for n JVMs? Async-profiler is unavailable and we have 10 JVMs, this is printed 10 times?

Reply:
Yes, I think we do want to fail here. If not, then we should break out and remove the datatype on the first error.

Reply:
What is the output on a panic? We want to fail gracefully.

Reply:
This is the output when asprof is not downloaded:

> thread 'main' panicked at src/data/java_profile.rs:53:25:
'asprof' command not found.
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

Reply:
Can we return an Error defined in PDError? That should force aperf to remove the java collector.

Reply:
This would allow aperf to continue right? Should we keep the panics on missing jps/pgrep?

Reply:
Ah. I see. Yeah. We want to not continue.